### ISSUE TITLE
[no-implicit-any-catch] TS 4.4 --useUnknownInCatchVariables

### ISSUE THREAD

**Issue by danielrentz** [typescript-eslint/typescript-eslint#3886]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-implicit-any-catch": "error"
  }
}
```

```TS
try { } catch (err) { }
```

```JSON
{
    "$schema": "https://json.schemastore.org/tsconfig",
    "compilerOptions": {
        "strict": true
    }
}
```

**Expected Result**

No linter error. Rule should check compiler configuration for existence of `useUnknownInCatchVariables` flag (or implicitly via `strict`) to decide whether to trigger for missing type.

**Actual Result**

Rule triggers with given settings.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.31.0` |
| `@typescript-eslint/parser`        | `4.31.0` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.3` |

**Comment by bradzacher** [comment]
the rule simply ensures you have an annotation because pre 4.4 an unannotated catch variable would be `any`.

Post 4.4 with the option turned on there's no longer any need for the rule - so you can and should disable it.
The rule is being deprecated in the next major.

**Comment by danielrentz** [comment]
Even better, thanks!
